### PR TITLE
CLOUDP-108872: Create one private endpoint interface.

### PR DESCRIPTION
**Pull request by johansteyn**
## Proposed changes

_Jira ticket:_ CLOUDP-108872

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

I have "smoke-tested" this locally using:

`      % mongocli atlas privateEndpoints gcp interfaces create endpoint-johan-c --endpointServiceId 61f3ce024e350b6457185d73 --gcpProjectId mcli-private-endpoints --endpoint endpoint-johan-c-0@10.128.0.2,endpoint-johan-c-1@10.128.0.3,endpoint-johan-c-2@10.128.0.4,endpoint-johan-c-3@10.128.0.5,endpoint-johan-c-4@10.128.0.6,endpoint-johan-c-5@10.128.0.7`

### CODE REVIEW

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+package interfaces
+
+import (
+	"context"
+	"strings"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/mongodb/mongocli/internal/cli"
```
Comment (andreaangiolillo) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:23]: please, remove the new line and run `make fmt`

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.privateEndpointGroupID = args[0]
+			return opts.Run()
+		},
+	}
+	cmd.Flags().StringVar(&opts.privateEndpointServiceID, flag.EndpointServiceID, "", usage.EndpointServiceID)
+	cmd.Flags().StringVar(&opts.gcpProjectID, flag.GCPProjectID, "", usage.GCPProjectID)
+	cmd.Flags().StringSliceVar(&opts.endpoints, flag.Endpoint, []string{}, usage.LinkTokenAccessListCIDREntries)
```
Comment (andreaangiolillo) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:109]:
```suggestion
	cmd.Flags().StringSliceVar(&opts.endpoints, flag.Endpoint, []string{}, usage.Endpoint)
```
The usage is wrong. Please, create the correct usage for this flag. It should contains an example with the separator `@`

**Review by andreaangiolillo — COMMENTED** [review]
Great work. Just a few comments

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go (hunk 1 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/create.go]: OK, that has been done with the latest commit.

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go (hunk 2 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/create.go]: Copy-paste mistake... thanks for spotting that! I have fixed it in the latest commit.

File: internal/usage/usage.go [hunk 3]
```diff
@@ -237,6 +237,7 @@ const (
 	EndpointServiceID                        = "Unique identifier of the private endpoint service for which you want to retrieve a private endpoint."
 	PrivateEndpointIDAzure                   = "Unique identifier of the Azure private endpoint resource."
 	PrivateEndpointIPAddressAzure            = "Private IP address of the private endpoint network interface you created in your Azure VNet."
+	Endpoint                                 = "List of GCP endpoints in the group, of form: endpointName1@ipAddress1,...,endpointNameN@ipAddressN"
```
Comment (JuliaMongo) [inline internal/usage/usage.go:240]: List of GCP endpoints in the group separated by commas, such as:

**Review by JuliaMongo — DISMISSED** [review]
LGTM with one small edit.

File: internal/usage/usage.go (hunk 3 above)
Comment (johansteyn) [reply internal/usage/usage.go]: Thanks Julia, I have changed that in the latest commit now.

**Review by JuliaMongo — APPROVED** [review]

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *CreateOpts) parseEndpoints() []*atlas.GCPEndpoint {
+	endpoints := make([]*atlas.GCPEndpoint, len(opts.endpoints))
+	for i, endpoint := range opts.endpoints {
+		s := strings.Split(endpoint, "@")
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:52]: [nit] let's please validate this is correct, `if len(s) !=2` kind of thing, other wise if smaller you'll get a NPE when accessing `[1]`

File: internal/usage/usage.go [hunk 5]
```diff
@@ -237,6 +237,7 @@ const (
 	EndpointServiceID                        = "Unique identifier of the private endpoint service for which you want to retrieve a private endpoint."
 	PrivateEndpointIDAzure                   = "Unique identifier of the Azure private endpoint resource."
 	PrivateEndpointIPAddressAzure            = "Private IP address of the private endpoint network interface you created in your Azure VNet."
+	Endpoint                                 = "List of GCP endpoints in the group separated by commas, such as: endpointName1@ipAddress1,...,endpointNameN@ipAddressN"
```
Comment (gssbzn) [inline internal/usage/usage.go:240]: [nit] it doesn't have to be separated by commas `--endpoint endpoint-0@10.142.0.2,endpoint-1@10.142.0.3` and `--endpoint endpoint-0@10.142.0.2 --endpoint endpoint-1@10.142.0.3` are both acceptable ways, cc @JuliaMongo

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go (hunk 4 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:52]: Agreed. I'll add a function to validate the individual endpoint strings...

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+			return opts.Run()
+		},
+	}
+	cmd.Flags().StringVar(&opts.privateEndpointServiceID, flag.EndpointServiceID, "", usage.EndpointServiceID)
+	cmd.Flags().StringVar(&opts.gcpProjectID, flag.GCPProjectID, "", usage.GCPProjectID)
+	cmd.Flags().StringSliceVar(&opts.endpoints, flag.Endpoint, []string{}, usage.Endpoint)
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
+	_ = cmd.MarkFlagRequired(flag.EndpointServiceID)
+	_ = cmd.MarkFlagRequired(flag.GCPProjectID)
```
Comment (johansteyn) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:112]: Question for team mates @andreaangiolillo @gssbzn : Surely we don't need to assign to the underscore?

ie. we could replace:
`	_ = cmd.MarkFlagRequired(flag.GCPProjectID)
`with simply:
`	cmd.MarkFlagRequired(flag.GCPProjectID)
`
(the code I have here was a straight copy-paste form the AWS code)

Comment (colm-quinn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:112]: I think that's a go thing - unassigned return values are caught as an error?

Comment (gssbzn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:112]: `MarkFlagRequired` returns an error if the flag doesn't exists in the set of flags of the command, we should either check it or explicitly ignore it

Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:112]: It compiles fine when I remove the assignment, so the language allows leaving it out.

But I get what Gustavo says: by assigning the returned error to underscore we are being more explicit - we are saying that we are aware of a potential error and deliberately ignoring it. 

So,  I'll keep it as-is, thanks!

File: internal/cli/atlas/privateendpoints/gcp/interfaces/create.go (hunk 4 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/create.go:63]: OK, I have added a new validateEndpoints function as well as a unit test for it.

**Pull request merged**